lib/srv/server/ec2_watcher.go

```diff
@@ -104,8 +131,30 @@ type ec2InstanceFetcher struct {
 	Region       string
 	DocumentName string
 	Parameters   map[string]string
+
+	// foundInstances keeps all of the ec2 instances that were matched
+	// in the last run of GetInstances for use as a cache with
+	// GetMatchingInstances
+	foundInstances map[foundInstancesKey]struct{}
```

Comment:
Nit: Not really a fan of "foundInstances" name, can we rename to
```suggestion
	cachedInstances map[cachedInstanceKey]struct{}
```
?